flow/connectors/core.go

```diff
@@ -290,6 +397,15 @@ func GetAs[T Connector](ctx context.Context, config *protos.Peer) (T, error) {
 	}
 }
 
+func GetByNameAs[T Connector](ctx context.Context, catalogPool *pgxpool.Pool, name string) (T, error) {
```

Comment:
this wouldn't play well with `Edit Peer`, we need to cache peer data atleast across one sync+normalize or QRep run to avoid peers change in the middle of a flow

Reply:
Guess Edit Peer requires pausing all mirrors for that peer then